docs/v3/core/index.rst

```diff
@@ -1598,38 +1520,125 @@ Storage transformers may be stacked to combine different functionalities:
 
 .. _extensions_section:
 
-Extension points
-================
+Extensions
+==========
 
+This section describes how additional functionality can defined
+for Zarr datasets by extended the `metadata documents`_.
+
+Extension points
+----------------
 
 Different types of extensions can exist and they can be grouped as follows:
 
-=========== ======================= ================================================
-level       extension               metadata
-=========== ======================= ================================================
-array       data type               `data_type`_
-array       chunk grid              `chunk_grid`_
-array       chunk key encoding      `chunk_key_encoding`_
-array       codecs                  `codecs`_
-array       storage transformer     `storage_transformers`_
-=========== ======================= ================================================
+=========== ======================= ====================================== ================================
+level       extension               metadata                               core definitions
```

Comment:
this applies to the old text, but I find these columns confusing. array is a node type, not a "level". I also don't really see the point of the "extension" column -- presumably the name of the extension point is just the relevant field in array metadata? And I'm not sure why we need the "metadata column". I think better columns would be

| node | field |
|-|-|
| array | [`data_type`](`data_type`_) |

i.e., we indicate the node type (array or group) and the name of the field, which can be a hyperlink to the actual specification for that field